src/main/scala/sbtassembly/Assembly.scala

```diff
@@ -291,9 +297,16 @@ object Assembly {
 
   private[sbtassembly] def sha1 = MessageDigest.getInstance("SHA-1")
   private[sbtassembly] def sha1content(f: File): String  = bytesToSha1String(IO.readBytes(f))
-  private[sbtassembly] def sha1name(f: File): String     = sha1string(f.getCanonicalPath)
+  private[sbtassembly] def sha1name(f: File)(implicit ao: AssemblyOption): String = sha1SafeString(f.getCanonicalPath)
```

Comment:
A lot cleaner to write `sha1name(jar.data) + "_" + sha1content(jar.data) + "_" + sha1rules(jarRules)` than `sha1name(jar.data, ao) + "_" + sha1content(jar.data) + "_" + sha1rules(jarRules, ao)`

This apply to some other places around the file.